modules/kafka/src/main/java/org/testcontainers/containers/KafkaContainer.java

```diff
@@ -80,83 +70,59 @@ public KafkaContainer withExternalZookeeper(String connectString) {
     }
 
     public String getBootstrapServers() {
-        if (port == PORT_NOT_ASSIGNED) {
-            throw new IllegalStateException("You should start Kafka container first");
-        }
-        return String.format("PLAINTEXT://%s:%s", getHost(), port);
-    }
-
-    @Override
-    protected void doStart() {
-        withCommand("sh", "-c", "while [ ! -f " + STARTER_SCRIPT + " ]; do sleep 0.1; done; " + STARTER_SCRIPT);
-
-        if (externalZookeeperConnect == null) {
-            addExposedPort(ZOOKEEPER_PORT);
-        }
-
-        super.doStart();
+        return String.format("PLAINTEXT://%s:%s", getHost(), getMappedPort(KAFKA_PORT));
     }
 
     @Override
-    @SneakyThrows
-    protected void containerIsStarting(InspectContainerResponse containerInfo, boolean reused) {
-        super.containerIsStarting(containerInfo, reused);
-
-        port = getMappedPort(KAFKA_PORT);
-
-        if (reused) {
-            return;
-        }
+    protected void configure() {
+        withEnv(
+            "KAFKA_ADVERTISED_LISTENERS",
+            String.format(
+                "BROKER://%s:9092",
+                getNetwork() != null
+                    ? getNetworkAliases().get(0)
+                    : "localhost"
+            )
+        );
 
         String command = "#!/bin/bash\n";
-        final String zookeeperConnect;
         if (externalZookeeperConnect != null) {
-            zookeeperConnect = externalZookeeperConnect;
+            withEnv("KAFKA_ZOOKEEPER_CONNECT", externalZookeeperConnect);
         } else {
-            zookeeperConnect = "localhost:" + ZOOKEEPER_PORT;
+            addExposedPort(ZOOKEEPER_PORT);
+            withEnv("KAFKA_ZOOKEEPER_CONNECT", "localhost:" + ZOOKEEPER_PORT);
             command += "echo 'clientPort=" + ZOOKEEPER_PORT + "' > zookeeper.properties\n";
             command += "echo 'dataDir=/var/lib/zookeeper/data' >> zookeeper.properties\n";
             command += "echo 'dataLogDir=/var/lib/zookeeper/log' >> zookeeper.properties\n";
             command += "zookeeper-server-start zookeeper.properties &\n";
         }
 
-        command += "export KAFKA_ZOOKEEPER_CONNECT='" + zookeeperConnect + "'\n";
-
-        command += "export KAFKA_ADVERTISED_LISTENERS='" + String.join(",", getBootstrapServers(), brokerAdvertisedListener(containerInfo)) + "'\n";
-
-        command += ". /etc/confluent/docker/bash-config \n";
-        command += "/etc/confluent/docker/configure \n";
-        command += "/etc/confluent/docker/launch \n";
+        // Optimization: skip the checks
+        command += "echo '' > /etc/confluent/docker/ensure \n";
+        // Run the original command
+        command += "/etc/confluent/docker/run \n";
+        withCommand("sh", "-c", command);
```

Comment:
this change is a tiny bit unrelated but, after looking for ways of simplifying things, I figured that we could delegate to the original `CMD` of Confluent's Kafka image *but* optimize it by removing the `ensure` step that adds ~2s, as it needs to instantiate a ZooKeeper client (not really necessary in our case, we have our own checks)

Reply:
I really like this change.